cmd/agent/app/httpserver/tchannel/manager.go

```diff
@@ -12,14 +12,15 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package httpserver
+package tchannel
```

Comment:
I'm not a big fan of the new package structure: without looking at the code, I would expect that the packages `cmd/agent/app/httpserver/[tchannel|grpc]` would contain code that ties something about gRPC or TChannel with the HTTP server that the agent starts, when in reality, all these packages have are specific implementations of the `ClientConfigManager` interface.

Perhaps the managers deserve their own packages? `cmd/agent/app/clientconfigmanager/[tchannel|grpc]`? Perhaps they should even be outside of the `cmd` package.

Reply:
The manager interface is used only in httpserver package Therefore it was originally colocated there I am fine scoping it under httpserver.

> Perhaps they should even be outside of the cmd package.

This applies also to reporter and probably other subpackages. It would make sense to move them  if they could be reused in other cmd. This is specific to the agent only.